### PR TITLE
Use importlib_metadata for Python < 3.10.2

### PR DESCRIPTION
**Pull request by GianlucaFicarelli**
Closes https://github.com/pypa/build/issues/692.

### CODE REVIEW

File: src/build/_importlib.py [hunk 1]
```diff
@@ -1,13 +1,8 @@
 import sys
 
 
-if sys.version_info < (3, 8):
+if sys.version_info < (3, 10, 2):
     import importlib_metadata as metadata
-elif sys.version_info < (3, 9, 10) or (3, 10, 0) <= sys.version_info < (3, 10, 2):
```
Comment (henryiii) [inline src/build/_importlib.py:6]: Please don't remove the try/except part. That helps bootstrapping (where the `.json` property isn't required).

File: pyproject.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ dependencies = [
   "pyproject_hooks",
   # not actually a runtime dependency, only supplied as there is not "recommended dependency" support
   'colorama; os_name == "nt"',
-  'importlib-metadata >= 4.6; python_version < "3.10"',  # Not required for 3.8+, but fixes a stdlib bug
+  'importlib-metadata >= 4.6; python_version < "3.10.2"',  # Not required for 3.8+, but fixes a stdlib bug
```
[the remark this answers is not in the document]
Comment (GianlucaFicarelli) [reply pyproject.toml]: Thanks 👍  Just for the record the correct name is `python_full_version` according to https://packaging.python.org/en/latest/specifications/dependency-specifiers/

File: src/build/_importlib.py (hunk 1 above)
Comment (GianlucaFicarelli) [reply src/build/_importlib.py:6]: I added the try/except and a comment with the bootstring. Do you think it's correct and understandable?

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (henryiii) [reply pyproject.toml]: Oops, yes, I don't use it very often. :)

**Review by henryiii — APPROVED** [review]

File: src/build/_importlib.py [hunk 3]
```diff
@@ -1,14 +1,13 @@
 import sys
 
 
-if sys.version_info < (3, 8):
-    import importlib_metadata as metadata
-elif sys.version_info < (3, 9, 10) or (3, 10, 0) <= sys.version_info < (3, 10, 2):
+if sys.version_info >= (3, 10, 2):
+    from importlib import metadata  # type: ignore[attr-defined]
```
Comment (GianlucaFicarelli) [inline src/build/_importlib.py:5]: The `type: ignore` workaround is needed to pass the "type" job https://github.com/pypa/build/actions/runs/6558894339/job/17813388696?pr=693
I didn't find a better way 🤔

Comment (henryiii) [reply src/build/_importlib.py:5]: Something like this?

```python
if sys.version_info < (3, 8):
    import importlib_metadata as metadata
elif sys.version_info >= (3, 10, 2):
    from importlib import metadata
else:
    try:
        import importlib_metadata as metadata
    except ModuleNotFoundError:
        # helps bootstrapping when dependencies aren't installed
        from importlib import metadata
```

Comment (GianlucaFicarelli) [reply src/build/_importlib.py]: Yes this worked, thanks (side note, I'm surprised that mypy still raises an error when the the order of the first two conditions is inverted)

**Review by henryiii — APPROVED** [review]

**Pull request merged**